consensus/validation.go

```diff
@@ -51,6 +51,16 @@ func (v *defaultMessageValidator) Validate(ctx context.Context, msg *types.Signe
 		return errNonAccountActor
 	}
 
+	if msg.Value.LessThan(types.ZeroAttoFIL) {
+		log.Info("Cannot transfer negative value", fromActor, msg.Value)
+		return errNegativeValue
+	}
+
+	if msg.GasLimit > types.BlockGasLimit {
```

Comment:
Similar to the above comment, this check is also performed here: https://github.com/filecoin-project/go-filecoin/blob/25b666da7e07f0a8b9f739ad2af2eec5b149ec56/consensus/processor.go#L605

Reply:
@rkowalick This code is used both before sending and prior to processing. We don't want users to send messages that are going to fail because they are over the block limit. Also, messages with this problem should fail faster on the mining side, because we can't charge gas for these messages. The check in the gas tracker isn't hurting anything, but messages really should not have gotten that far.

Reply:
Since validate is called both in `Sender.send` and in `attemptApplyMessage`, the old "gas block limit" check is redundant and should be removed.